brave-spark-framework-filter/src/test/java/com/github/kristofa/brave/spark/ITBraveSparkTest.java

```diff
@@ -0,0 +1,66 @@
+package com.github.kristofa.brave.spark;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.http.ITHttpServer;
+import com.github.kristofa.brave.http.SpanNameProvider;
+import org.junit.After;
+import org.junit.AssumptionViolatedException;
+import org.junit.Test;
+import spark.ExceptionHandler;
+import spark.Request;
+import spark.Response;
+import spark.Spark;
+
+public class ITBraveSparkTest extends ITHttpServer {
+
+    /**
+     * after filter executed before NotFoundRoute,there is no way to add 404 code after serverspan
+     * has been setServerSend,may be throw a not found exceeption and process it in ExceptionHandler
+     */
+    @Override
+    @Test
+    public void addsStatusCodeWhenNotOk() throws Exception {
+        throw new AssumptionViolatedException("Filters run before NotFoundRoute can be determined");
+    }
+    @Override
+    protected void init(Brave brave, SpanNameProvider spanNameProvider) throws Exception {
+        stop();
+        Spark.before(BraveSparkFilter.builder(brave).spanNameProvider(spanNameProvider).build().requestFilter());
+        Spark.get("/foo", (req, res) -> "bar");

```

Comment:
please move the route definitions below the brave setup code, as eventhough it is valid, it is confusing to read a mix of concerns.